packages/dart/npt_flutter/lib/features/onboarding/util/profile_progress_listener.dart

```diff
@@ -4,18 +4,74 @@ import 'dart:developer';
 import 'package:at_client_mobile/at_client_mobile.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:npt_flutter/app.dart';
+import 'package:npt_flutter/constants.dart';
 import 'package:npt_flutter/features/profile_list/bloc/profile_list_bloc.dart';
+import 'package:npt_flutter/util/uuid.dart';
 
 import '../../profile_list/cubit/sync_cubit.dart';
 
 class ProfileProgressListener extends SyncProgressListener {
+  ProfileProgressListener(AtClient atClient) {
+    Stream<AtNotification> subscription =
+        atClient.notificationService.subscribe(
+            regex: '\\.${Uuid.profilesSubNamespace}'
+                '\\.${Constants.namespace!}@');
+    subscription.listen((AtNotification n) async {
+      try {
+        final profileListBlock =
+            App.navState.currentContext!.read<ProfileListBloc>();
+
+        App.log('[INFO] Notification $n'.loggable);
+        if (n.key.contains(
+            '.${Uuid.profilesSubNamespace}.${Constants.namespace!}')) {
+          switch (n.operation) {
+            case 'update':
+              atClient.syncService.sync();
+              break;
+
+            case 'delete':
+              await atClient
+                  .getLocalSecondary()!
+                  .keyStore!
+                  .remove('cached:${n.key}', skipCommit: true);
+              profileListBlock.add(const ProfileListLoadEvent());
+              log('ProfileProgressListener: ProfileListLoadEvent triggered to reload profiles');
+              break;
+
+            default:
+              break;
+          }
+        }
+      } catch (ignore) {
+        App.log('[ERROR] Caught $ignore while listening to notifications.'
+                ' Ignoring and continuing.'
+            .loggable);
+      }
+    });
+  }
+
   @override
   void onSyncProgressEvent(SyncProgress syncProgress) async {
     unawaited(App.navState.currentContext!.read<SyncCubit>().checkSync());
-    final profileListBlock = App.navState.currentContext!.read<ProfileListBloc>();
+    final profileListBlock =
+        App.navState.currentContext!.read<ProfileListBloc>();
+
+    bool profileSynced() {
+      for (final KeyInfo ki in syncProgress.keyInfoList ?? []) {
+        App.log('[DEBUG] sync keyInfo: $ki'.loggable);
+        if (ki.syncDirection == SyncDirection.remoteToLocal) {
+          if (ki.key.contains(
+              '.${Uuid.profilesSubNamespace}.${Constants.namespace!}')) {
+            return true;
+          }
+        }
+      }
+      return false;
+    }
 
     if (syncProgress.syncStatus == SyncStatus.success &&
-        (profileListBlock.state is ProfileListLoaded &&
+        profileListBlock.state is ProfileListLoaded &&
+        (profileSynced() ||
```

Comment:
Now triggering the ProfileListLoadEvent not just on the initial sync of the app, but on subsequent syncs